src/frontend/src/components/modal.ts

```diff
@@ -0,0 +1,59 @@
+import { html, TemplateResult, render } from "lit-html";
+
+export const createModal = ({
+  slot,
+  id,
+  confirmText = "Ok",
+}: {
+  slot: TemplateResult;
+  id?: string;
+  confirmText?: string;
+}): HTMLDialogElement => {
+  const modalHtml = html`
+    <form method="dialog" class="c-modal__content c-card c-card--modal">
+      <div class="c-modal__inner">
+        <span class="c-modal__close" data-closemodal>&times;</span>
+        ${slot}
+        <div class="c-modal__footer">
+          <button type="submit" class="c-button c-button--primary">
+            ${confirmText}
+          </button>
+        </div>
+      </div>
+    </form>
+  `;
+
+  const $modal = document.createElement("dialog");
+  $modal.id = id != null ? id : "modal";
+  $modal.classList.add("c-modal");
+  $modal.setAttribute("aria-modal", "true");
+
+  render(modalHtml, $modal);
+  return $modal;
+};
+
+export const modal = (slot: TemplateResult): Promise<void> => {
+  // create modal and append it to the body
+  const $modal = createModal({ slot });
+  document.body.appendChild($modal);
+
+  return new Promise((resolve) => {
+    // open modal
+    $modal.showModal();
+    const $modalClose = $modal.querySelector(
```

Comment:
would it be useful to use a mutationobserver to be sure those buttons are added to the dom? you are casting to HTMLElement but they can be null and throw an exception when the onclick is attached

Reply:
not sure how a mutation observer would solve that? can you make an example?

Reply:
Just noticing this comment of @nmattia https://github.com/dfinity/internet-identity/pull/1306/files#r1128004572 which per extension makes my comment obsolte since queries will disappear

Reply:
thanks for caring though!